x/staking/keeper/weight.go

```diff
@@ -13,10 +14,17 @@ const (
 	blockTimestampEpoch = int64(946684800)
 	secondsPerWeek      = int64(60 * 60 * 24 * 7)
 	weeksPerYear        = float64(52)
+	fixedTimeStamp      = int64(1685577600) // 2023-06-01 00:00:00 GMT+0
 )
 
-func calculateWeight(nowTime int64, tokens sdk.Dec) (shares types.Shares, sdkErr error) {
-	nowWeek := (nowTime - blockTimestampEpoch) / secondsPerWeek
+func calculateWeight(nowTime int64, tokens sdk.Dec, height int64) (shares types.Shares, sdkErr error) {
+	var nowWeek int64
+	if types2.HigherThanVenus6(height) {
```

Comment:
why change this func， i think it‘s dangerous，can chang staking revward

Reply:
> why change this func， i think it‘s dangerous，can chang staking revward

other pr https://github.com/okx/exchain/pull/3108 handle this func